src/librustc/traits/specialize/mod.rs

```diff
@@ -103,6 +105,38 @@ pub fn translate_substs<'a, 'gcx, 'tcx>(infcx: &InferCtxt<'a, 'gcx, 'tcx>,
     source_substs.rebase_onto(infcx.tcx, source_impl, target_substs)
 }
 
+/// Locates the applicable definition of a method, given its name.
```

Comment:
This comment doesn't seem sufficient. What is `impl_def_id` and `impl_substs`, for example? What is `substs` and how does it relate to `impl_substs`? 

Probably nice to give some context about where/why it is used, too.